gemrb/core/Streams/DataStream.cpp

```diff
@@ -123,35 +123,28 @@ strret_t DataStream::ReadRegion(Region& r)
 	return ret;
 }
 
-strret_t DataStream::ReadLine(void* buf, strpos_t maxlen)
+strret_t DataStream::ReadLine(std::string& buf, strpos_t maxlen)
 {
-	// FIXME: eof?
-	if (!maxlen) {
-		return 0;
-	}
-	unsigned char * p = ( unsigned char * ) buf;
 	if (Pos >= size) {
-		p[0]=0;
 		return Error;
 	}
-	unsigned int i = 0;
+	
+	buf.clear();
+	strpos_t i = 0;
 	//TODO: fix this to handle any combination of \r and \n
 	//Windows: \r\n
 	//Old Mac: \r
 	//otherOS: \n
-	while (i < ( maxlen - 1 )) {
+	while (Pos < size && i < (maxlen - 1)) {
 		char ch;
-		Read(&ch, 1);
+		i += Read(&ch, 1);
 		if (ch == '\n')
 			break;
```

Comment:
Not sure if it matters, but this case now returns a value 1 greater than before.

Reply:
yeah, it should be fine. I think we only check it for `Error` and use `string::length()` for the actual data examination. It should be returning the number of bytes read which will always be longer.